pages/books.tsx

```diff
@@ -34,6 +34,19 @@ const Catalog = () => {
 
   const { response } = jsonData.data;
 
+  //search query filters based on all fields, with memoization
+  const filteredItems = useMemo(() => {
+    const keys: string[] = Object.keys(response[0]);
```

Comment:
Can rewrite this a bit simpler:

```
return response.filter((item) => {
  return Object.values(item).includes(new RegExp(`/${searchValue}/i`););
}
```

(idk if that regex is exactly right you may need the `i` as second parameter, makes it case insensitive)

Reply:
I played around with the syntax a bit and got it working with regex: 

```
    return response.filter((item) => {
      return new RegExp(searchValue, 'i').test(Object.values(item).toString());
    });
```